src/h5web/explorer/utils.ts

```diff
@@ -60,38 +60,36 @@ function buildDatatype(
 function buildGroup(
   metadata: HDF5Metadata,
   link: HDF5HardLink | HDF5RootLink,
-  parents: MyHDF5Group[]
+  parent?: MyHDF5Group
 ): MyHDF5Group {
   const rawGroup = metadata.groups[link.id];
 
   const group: MyHDF5Group = {
-    ...prepareEntity(link, parents),
+    ...prepareEntity(link, parent),
     kind: MyHDF5EntityKind.Group,
     attributes: rawGroup.attributes || [],
     children: [],
   };
 
-  const newParents = [...parents, group];
-
   group.children = (rawGroup.links || []).map((link) => {
     if (!isReachable(link)) {
       return {
         uid: nanoid(),
         name: link.title,
         kind: MyHDF5EntityKind.Link,
-        parents: newParents,
+        parent: group,
         attributes: [],
         rawLink: link,
       };
     }
 
     switch (link.collection) {
       case HDF5Collection.Groups:
-        return buildGroup(metadata, link, newParents);
+        return buildGroup(metadata, link, group);
       case HDF5Collection.Datasets:
-        return buildDataset(metadata, link, newParents);
+        return buildDataset(metadata, link, group);
       default:
```

Comment:
I caught a glimpse of this before: I think the `default` of the switch should throw an error and that a `case: HDF5Collection.Datatypes` should be added so that creation of malformed entities do not fail silently.